### PR TITLE
Load all headers 

### PR DESCRIPTION
**Pull request by profxj**
Save a copy of all headers when loaded to generate fitsdict

There is probably a cleaner way to save these then the current
approach.  But that can be refactored later.

Checked on development suite.
Added a unit test too.

### CODE REVIEW

File: pypit/armlsd.py [hunk 1]
```diff
@@ -26,7 +26,7 @@
 msgs = armsgs.get_logger()
 
 
-def ARMLSD(fitsdict, reuseMaster=False, reloadMaster=True):
+def ARMLSD(fitsdict, allhead, reuseMaster=False, reloadMaster=True):
```
Comment (rcooke-ast) [inline pypit/armlsd.py]: We should add a similar key for armed.py

File: pypit/pypit.py [hunk 2]
```diff
@@ -222,7 +222,7 @@ def PYPIT(redname, debug=None, progname=__file__, quick=False, ncpus=1, verbosit
     if spect.__dict__['_spect']['mosaic']['reduction'] == 'ARMLSD':
         msgs.info("Data reduction will be performed using PYPIT-ARMLSD")
         from pypit import armlsd
-        status = armlsd.ARMLSD(fitsdict)
+        status = armlsd.ARMLSD(fitsdict, allhead)
     elif spect.__dict__['_spect']['mosaic']['reduction'] == 'ARMED':
         msgs.info("Data reduction will be performed using PYPIT-ARMED")
         from pypit import armed
```
Comment (rcooke-ast) [inline pypit/pypit.py]: and similarly on the line below too...

File: pypit/armlsd.py (hunk 1 above)
Comment (profxj) [inline pypit/armlsd.py]: added

File: pypit/pypit.py (hunk 2 above)
Comment (profxj) [inline pypit/pypit.py]: added

**Comment by rcooke-ast** [comment]
Is it worthwhile including all of the headers in fitsdict, instead of returning allhead as well? Something like:

fitsdict['headers'] = allhead

Admittedly, this information is stored (and accessed) differently from the other information in fitsdict, but perhaps this is somewhat tidier? I'm open to either the implementation you propose, or storing the headers in fitsdict as above.

**Comment by profxj** [comment]
I like your suggestion (and should have thought of it myself!).  Stand by..

**Comment by profxj** [comment]
Merging.

**Pull request merged**